types/ember/index.d.ts

```diff
@@ -1571,6 +1572,34 @@ declare namespace Ember {
             @return {Promise}
             */
             finally<V>(callback: (a: T) => V, label?: string): Promise<V, U>;
+
+            static all<Q, R>(promises: GlobalArray<(Q | Thenable<Q, R>)>): Promise<Q[], R>;
+            static race<Q, R>(promises: GlobalArray<Promise<Q, R>>): Promise<Q, R>;
+
+            /**
+             @method resolve
+             @param {Any} value value that the returned promise will be resolved with
+             @param {String} label optional string for identifying the returned promise.
+             Useful for tooling.
+             @return {Promise} a promise that will become fulfilled with the given
+             `value`
+             */
+            static resolve<Q, R>(object?: Q | Thenable<Q, R>): Promise<Q, R>;
+
+            /**
+             @method cast (Deprecated in favor of resolve
+             @param {Any} value value that the returned promise will be resolved with
+             @param {String} label optional string for identifying the returned promise.
+             Useful for tooling.
+             @return {Promise} a promise that will become fulfilled with the given
+             `value`
+             */
+            static cast<Q, R>(object: Q | Thenable<Q, R>, label?: string): Promise<Q, R>;
+
+            /**
+             `RSVP.Promise.reject` returns a promise rejected with the passed `reason`.
+             */
+            static reject(reason?: any): Promise<any, any>;
```

Comment:
I think this might work better if this is `static reject<R>(reason?: R): Promise<any, R>` – is there a reason I'm missing not to do that? (There very well might be; that's a genuine question.)